mlrun/projects/project.py

```diff
@@ -2306,6 +2306,19 @@ def create_remote(self, url, name="origin", branch=None):
         self.spec._source = self.spec.source or url
         self.spec.origin_url = self.spec.origin_url or url
 
+    def _ensure_git_repo(self):
+        if self.spec.repo:
+            return
+        context = self.context
+        git_dir_path = os.path.join(context, ".git")
```

Comment:
you already have path imported so you don't need to import os
```suggestion
        git_dir_path = path.join(context, ".git")
```